### PR TITLE
Timer

### PR DESCRIPTION
**Pull request by JamesAdamHughes**
A countdown for users to submit answers by. Master timer on the server will forcibly move everyone on at the end of the timer, and a client timer to give a (general) display of the amount of time left

****
MORE FEATURES HERE

### CODE REVIEW

File: client/Angular/controllers/mainController.js [hunk 1]
```diff
@@ -22,6 +22,25 @@ ClonageApp.controller("MainController", function($scope, userService, roomServic
...
+                $scope.counter = timerCount ;
+                timerCount-= 1 ;
+                $timeout(countDown, 1000);
+
+            }
+        };
+        countDown();
+    };
+
     errorService.registerErrorListener(displayErrorMessage)
```
Comment (JamesAdamHughes) [inline client/Angular/controllers/mainController.js]: Missing semicolons, will be picked up byJsHint

File: client/Angular/controllers/mainController.js [hunk 2]
```diff
@@ -22,6 +22,25 @@ ClonageApp.controller("MainController", function($scope, userService, roomServic
         toastr.error(errorMessage);
     }
 
+    $scope.startCountdown = function(page) {
```
Comment (JamesAdamHughes) [inline client/Angular/controllers/mainController.js]: maybe comment this so we know this is just client side

File: client/Angular/services/gameService.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-ClonageApp.service('gameService', ['communicationService', function(communicationService) {
+ClonageApp.service('gameService', ['communicationService', '$timeout', function(communicationService, $timeout) {
```
Comment (JamesAdamHughes) [inline client/Angular/services/gameService.js]: $timeout is injected but never seems to be used?

File: server/GameController.js [hunk 4]
```diff
@@ -29,6 +28,55 @@ module.exports = function(data) {
...
+    GameState:
+        0 : game hasn't started yet
+        _1 : game started, round _, everyone has to submit a choice
+        _2 : round _, everyone has to submit a vote
+        _3 : round _, votes are in
+        4: final results, game over
+    */
+    var GameState = '0';
+    var GameStateHasChanged = false;
+	var count = 30;
```
Comment (JamesAdamHughes) [inline server/GameController.js]: fix indentation

File: server/server.js [hunk 5]
```diff
@@ -258,6 +258,7 @@ module.exports = function(port, enableLogging) {
             var room = getRoomFromId(roomId);
 
             room.gameController.newRound(function(data) {
+                console.log(data.gameIsOver);
```
Comment (JamesAdamHughes) [inline server/server.js]: can remove the console.log

File: server/server.js [hunk 6]
```diff
@@ -286,11 +287,29 @@ module.exports = function(port, enableLogging) {
                             }
                         });
                     });
+
+                    room.gameController.startTimer(function(data) {
```
Comment (JamesAdamHughes) [inline server/server.js]: comment this to be explicit?

File: server/GameController.js [hunk 7]
```diff
@@ -29,6 +28,55 @@ module.exports = function(data) {
...
+
+	var callBackReference = null;
+
+    var startTimer = function(callback) {
+
+    	callBackReference = callback;
+
+    	GameStateHasChanged = false;
+
+	    count = 30;
```
Comment (JamesAdamHughes) [inline server/GameController.js]: using count to show the round count (no of rounds) maybe change this to roundTimeLimit or something?

File: server/GameController.js [hunk 8]
```diff
@@ -29,6 +28,55 @@ module.exports = function(data) {
...
+        _3 : round _, votes are in
+        4: final results, game over
+    */
+    var GameState = '0';
+    var GameStateHasChanged = false;
+	var count = 30;
+
+	var callBackReference = null;
+
+    var startTimer = function(callback) {
```
Comment (JamesAdamHughes) [inline server/GameController.js]: comment?

**Comment by Geor23** [comment]
DO NOT MERGE!

File: server/GameController.js (hunk 8 above)
Comment (Geor23) [inline server/GameController.js]: a lot actually got changed as i merged ben's version into mine. It's not finished yet, I'll add comments once i get everything working and i push the final version

Comment (JamesAdamHughes) [inline server/GameController.js]: :)

File: server/GameController.js [hunk 9]
```diff
@@ -20,15 +21,72 @@ module.exports = function(data) {
...
+		which is outside of the scope of startTimer
+	*/
+	var counter;
+
+	// function to initialise the countdown
+    var startTimer = function(callback) {
+
+    	count = 30 ;
+
+	    timerIsActive = true;
```
Comment (JamesAdamHughes) [inline server/GameController.js]: indentation :)

File: server/GameController.js [hunk 10]
```diff
@@ -402,6 +455,58 @@ module.exports = function(data) {
...
+			});
+
+			voteNumber = 0 ;
+
+			// Update every player's rank in the room
+			setRank();
+
+		}
+
+		 /*else if (GameState === POSSIBLE_GAMESTATES.VOTING) {
```
Comment (JamesAdamHughes) [inline server/GameController.js]: take out this commented bit if we don't need it

**Comment by JamesAdamHughes** [comment]
okay with those changes

File: client/includes/css/test-site.css [hunk 11]
```diff
@@ -180,6 +180,15 @@ body, html {
   margin-top: 60px;
 }
 
+.player-ready {
+  color : white;
```
Comment (JamesAdamHughes) [inline client/includes/css/test-site.css]: don't think we need this anymore

**Pull request merged**